pkg/v1/remote/image.go

```diff
@@ -221,6 +221,16 @@ func (rl *remoteImageLayer) Descriptor() (*v1.Descriptor, error) {
 	return partial.BlobDescriptor(rl, rl.digest)
 }
 
+// See partial.Exists.
+func (rl *remoteImageLayer) Exists() (bool, error) {
+	resp, err := rl.ri.headBlob(rl.digest)
```

Comment:
If the method returns `(bool, error)` I would expect there to be some possibility of returning `false, nil` -- otherwise, the bool is just equivalent `err != nil`, and doesn't provide any other information.

Could this `return false, nil` if `headBlob` returns a 404? (Indicating that it _successfully_ determined that the blob _doesn't exist_)